docs/tutorials/getting-started-with-csharp/microservices.md

```diff
@@ -0,0 +1,443 @@
+scale the installation of your application.
+
+All the code in this tutorial will work in any .NET Core environment.
+The additional tasks for a Docker installation will work for an ASP.NET
+Core application. 
+
+# Prerequisites
+You’ll need to setup your machine to run .NET core. Instructions are
+[here](http://dotnet.github.io/getting-started/). You can run this
+application on Windows, Ubuntu Linux, OS X or in a Docker container. 
+You’ll need to install your favorite code editor. The descriptions below
+use [Visual Studio Code](https://code.visualstudio.com/) which is an open
+source, cross platform editor. However, you can use whatever tools you are
+comfortable with.
+
+You'll also need to install the Docker engine. Instructions for that are
+located [here](https://docs.docker.com/engine/installation/). Docker can
+be installed in many Linux distributions, Mac OS, or Windows. The page
+referenced above contains links to each of the available installations.
+
+You'll also need to install a number of command line tools that support
+asp.net core development. The command line templates use yeoman, bower,
+grunt, and gulp. You may already have many of these tools, but if not,
+run the following command in your favorite shell:
+
+`npm install -g yo bower grunt-cli gulp`
+
+This instructs the node package manager (npm) to install the needed tools.
+The '-g' option indicates that it is a global install, and those tools are
+available system wide. (A local install scopes the package to a single
+project). Once you've installed those core tools, you need to install
+the yeoman asp.net template generators:
+
+Then, you can install ASP.NET template generator:
+
+`npm install -g generator-aspnet`
+
+# Create the Application
+
+Now that you've installed all the tools, create a new asp.net core
+application. To use the command line generator, execute the following
+yeoman command in your favorite shell:
+
+`yo aspnet`
+
+This command prompts you to select what type of application you want to
+create. For this microservice, you want the simplest, most light weight
+web application possible, so select 'Empty Application'. The template
+will prompt you for a name. Select 'WeatherMicroservice'. 
+
+The template creates six files for you:
+
+* A .gitignore, customized for asp.net core applications.
+* A Startup.cs file. This contains the basis of the application.
+* A project.json file. This is the build file for the application.
+* A Dockerfile. This script creates a Docker image for the application.
+* A wwwroot/readme.md. This contains links to other asp.net core resources.
+* A wwwroot/web.config file. This contains basic configuration information.
+
+Let's run the template generated application. That's done using a series
+of tools from the command line. The first step is to restore
+all the dependencies:
+
+`dnu restore`
+
+Dnu restore uses the NuGet package manager to install all the necessary packages
```

Comment:
Should the reader know what "Dnu" or "dnu" is? Likewise with the "dnx" command below. It is a tutorial, so best to explain on first mention.